# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/firefox/hotfix-gcc14.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- thunderbird-128.0/media/ffvpx/libavcodec/av1dec.c.vanilla	2024-07-24 20:38:58.267914215 +0200
+++ thunderbird-128.0/media/ffvpx/libavcodec/av1dec.c	2024-07-24 20:39:43.530910543 +0200
@@ -887,7 +887,7 @@
         ff_cbs_fragment_reset(&s->current_obu);
     }
 
-    s->dovi.logctx = avctx;
+    s->dovi.logctx = (AVContext *)avctx;
     s->dovi.dv_profile = 10; // default for AV1
     sd = ff_get_coded_side_data(avctx, AV_PKT_DATA_DOVI_CONF);
     if (sd && sd->size > 0)
--- thunderbird-128.0/media/ffvpx/libavcodec/libdav1d.c.vanilla	2024-07-24 20:45:13.121883801 +0200
+++ thunderbird-128.0/media/ffvpx/libavcodec/libdav1d.c	2024-07-24 20:45:43.007881376 +0200
@@ -289,7 +289,7 @@
     c->delay = res > 1 ? res : 0;
 #endif
 
-    dav1d->dovi.logctx = c;
+    dav1d->dovi.logctx = (AVContext *)c;
     dav1d->dovi.dv_profile = 10; // default for AV1
     sd = ff_get_coded_side_data(c, AV_PKT_DATA_DOVI_CONF);
     if (sd && sd->size > 0)
